src/test/java/org/elasticsearch/index/query/TermQueryBuilderTest.java

```diff
@@ -0,0 +1,114 @@
+
+import org.apache.lucene.search.Filter;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.QueryWrapperFilter;
+import org.apache.lucene.search.TermQuery;
+import org.elasticsearch.common.lucene.BytesRefs;
+import org.elasticsearch.index.search.child.CustomQueryWrappingFilter;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+
+public class TermQueryBuilderTest extends BaseQueryTestCase<TermQueryBuilder> {
+
+    @Override
+    protected TermQueryBuilder createEmptyQueryBuilder() {
+        return new TermQueryBuilder();
+    }
+
+    /**
+     * @return a TermQuery with random field name and value, optional random boost and queryname
+     */
+    protected TermQueryBuilder createTestQueryBuilder() {
+        Object value = null;
+        switch (randomIntBetween(0, 3)) {
+        case 0:
+            value = randomBoolean();
+            break;
+        case 1:
+            if (randomInt(10) > 0) {
+                value = randomAsciiOfLength(8);
+            } else {
+                // generate unicode string in 10% of cases
+                value = randomUnicodeOfLength(10);
+            }
+            break;
+        case 2:
+            value = randomInt(10000);
+            break;
+        case 3:
+            value = randomDouble();
+            break;
+        }
+        TermQueryBuilder query = new TermQueryBuilder(randomAsciiOfLength(8), value);
+        if (randomBoolean()) {
+            query.boost(2.0f / randomIntBetween(1, 20));
+        }
+        if (randomBoolean()) {
+            query.queryName(randomAsciiOfLength(8));
+        }
+        return query;
+    }
+
+    @Override
+    protected void assertLuceneQuery(TermQueryBuilder queryBuilder, Query query, QueryParseContext context) throws IOException {
+        assertThat(query, instanceOf(TermQuery.class));
+        assertThat(query.getBoost(), is(queryBuilder.boost()));
+        TermQuery termQuery = (TermQuery) query;
+        assertThat(termQuery.getTerm().field(), is(queryBuilder.fieldName()));
+        assertThat(termQuery.getTerm().bytes(), is(BytesRefs.toBytesRef(queryBuilder.value())));
+        if (queryBuilder.queryName() != null) {
+            Filter namedQuery = context.copyNamedFilters().get(queryBuilder.queryName());
+            assertNotNull(namedQuery);
+            if (namedQuery instanceof CustomQueryWrappingFilter) {
+                assertThat(query, is(((CustomQueryWrappingFilter) namedQuery).getQuery()));
+            } else if (namedQuery instanceof QueryWrapperFilter) {
+                assertThat(query, is(((QueryWrapperFilter) namedQuery).getQuery()));
+            } else {
+                fail("Expected either a QueryWrapperFilter or a CustomQueryWrappingFilter to be registered under " + queryBuilder.queryName() +
+                        " in the query parse context");
+            }
+        }
+    }
+
```

Comment:
I think that's a good idea. However, this seems to be difficult in a generic way because there are many ways json can be invalid and the test setup needs to take implementation details into account (which exception will be thrown, which fieldnames are allowed. I tried this in the base class but ended up having to manipulate the query string in very crude ways. I'm going to open an issue for this so we can collect and discuss ideas on this. I think it doesn't have to go in with this PR.